ext/drcovlib/modules.c

```diff
@@ -234,7 +236,7 @@ static inline void
 lookup_helper_set_fields(module_entry_t *entry, OUT uint *mod_index, OUT app_pc *mod_base)
 {
     if (mod_index != NULL)
-        *mod_index = entry->id;
+        *mod_index = entry->containing_id; /* Yes, the main (containing) module. */
```

Comment:
I don't think this is what we want: drmodtrack treats the segments as separate entries and exposes that to the user.  The user expects the resulting index to actual contain the queried PC, which is not true with this change.  What is your reason for wanting to change this?